### DISCUSSION TITLE
Bad frame pacing on Android phone (S23 - Snap Dragon 8Gen2)

### DISCUSSION THREAD

**Discussion by TungAn** [ClassicOldSong/moonlight-android#8] [Q&A] [answered]
Hi,

I've been facing this issue for a long time when streaming the moonlight using an Android phone

My setup:
- Samsung Galaxy S23 Ultra, SOC is Snapdragon 8Gen2
- Moonlight Atermis 
- Screen resolution: 3088x1440
- Refresh rate: 120hz
- Bitrate / Decoding: 80mbps / HEVC
- Wifi 6 (I also test the connection to the host with iperf3, all good, no frame drop and the jitter is lower than recommended), host is PC connected via enthernet
- Frame pacing: Balanced, I cannot use the default "Prefer lowest latency" or other frame pacing due to stuttering
- My host is PC that can handle 120fps games well in 1440p (3090, 13900k and games are FC 24 or AC Mirage)
- Locked 120fps by Nvidia Panel, Vsync Off, Gsync Off. Using dummy plug to create the custom resolution that matched with the phone resolution and refresh rate

My symptom:
- I open the stream stats to see any weird number, but the statics showing good, no frame drop, no jitter, frame incoming, and rendering is stable at 120fps 
- If I use "Prefer lowest latency", I get the smooth frame (stable between 110 ~ 120fps) for 30sec or 1 min, then it drops between 90-100fps causing the terrible stuttering -> I monitored fps using GPU Watch settings (in Dev mode), then back to smooth again and keep stutter subsequently. I tried to turn off everything related to Wifi scanning, Bluetooth, and file sharing, and make the battery unrestricted but no luck
- Other frame pacing: "Balanced with FPS limit" or "Never drop frame" give me bad stuttering at all time
- However, when I use "Balanced", the stuttering is gone but latency is added. Judder is still there but less noticeable 

Do you guys have any idea? I want to enjoy the lowest latency possible but occasionally stutter driving me mad

Thanks for your help a

**Comment by ClassicOldSong** [comment]
Have you tried connect your phone with USB cable, enabled USB network sharing, then enter your PC ip assigned by your phone manually into Moonlight to see if it still stutters?

If it still happens, can you try streaming with another computer just to make sure it's not a computer issue? I have reports that Intel 13-14 gen CPUs might suffer from stuttering streams but AMD ones and older Intel platforms are just fine with the same client.

**Comment by TungAn** [comment]
thanks for your help.

Lemme try and report it back.

For the host, I only have an Intel PC, I will try to stream from the Legion Go instead since it AMD based.

Besides that, I also streamed from this PC by the Legion Go with the same setup, seems Legion Go handling frame pacing better (but it's quite heavy so I cannot hold it longer when laying on the bed compared to the phone)

The stuttering still there but gone when I enabled the v-sync option (moonlight-qt client has this) but yea latency is added

**Comment by TungAn** [comment]
[https://www.youtube.com/watch?v=bJdMZAN502c](https://youtu.be/bJdMZAN502c?si=FLJlbhCOVvOcFvYT&t=53) (at 1:00)

I tried USB network sharing, but still no luck. You can check the video above.

The rendering frame is consistent at 120fps, but the actual FPS is fluctuated

**Comment by ClassicOldSong** [comment]
I see, but I don't have a Samsung device and the GPU Watch seems to be a samsung specific thing.

This looks like a GPU driver issue from your phone but you still need to make sure that this happens to every host you're connecting. Slight stutter might happen when using lowest latency option but usually it wouldn't be this bad. Have you tried with different encodings like AV1?

Snapdragon 8G2 has hardware AV1 decoder, and I have evry good experience with it on my 8G3 devices, essentially cutting decoding latency half to 4ms when in 1440P.

**Comment by TungAn** [comment]
thanks for your support. yup Snap 8Gen2 supports AV1, but the decoding time is worse than HEVC (+5ms latency), also I suffer the same stuttering issue

I also tried to stream from Legion Go which is AMD-based, but no luck. I guess it is device-related or OS (Samsung One UI) related. You can see the vid below

[https://www.youtube.com/watch?v=fEF_mNzC8mc](https://www.youtube.com/watch?v=fEF_mNzC8mc)

Could you share the device model that you are using? maybe I will try to switch to that

**Reply by ClassicOldSong** [reply, marked as answer]
I'm using a Chinese local brand called Meizu

OnePlus should also do fine since I have no such issues on my OnePlus 8G3 tablet

But if you want the smoothest experience, just use iPhone

**Reply by TungAn** [reply]
Many thanks. Will consider the above
